db/config.cc

```diff
@@ -861,7 +861,7 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     , developer_mode(this, "developer_mode", value_status::Used, DEVELOPER_MODE_DEFAULT, "Relax environment checks. Setting to true can reduce performance and reliability significantly.")
     , skip_wait_for_gossip_to_settle(this, "skip_wait_for_gossip_to_settle", value_status::Used, -1, "An integer to configure the wait for gossip to settle. -1: wait normally, 0: do not wait at all, n: wait for at most n polls. Same as -Dcassandra.skip_wait_for_gossip_to_settle in cassandra.")
     , force_gossip_generation(this, "force_gossip_generation", liveness::LiveUpdate, value_status::Used, -1 , "Force gossip to use the generation number provided by user")
-    , experimental(this, "experimental", value_status::Used, false, "[Deprecated] Set to true to unlock all experimental features (except 'consistent-topology-changes' feature, which should be enabled explicitly via 'experimental-features' option). Please use 'experimental-features', instead.")
+    , experimental(this, "experimental", value_status::Unused, false, "[Deprecated] Set to true to unlock all experimental features (except 'consistent-topology-changes' feature, which should be enabled explicitly via 'experimental-features' option). Please use 'experimental-features', instead.")
```

Comment:
I wonder if instead of keeping "Unused" options, we shouldn't have just removed them.
Yes, it can cause upgrade to fail with an unrecognized option. But on the other hand, how is it better for the upgrade to nominally succeed and then it doesn't work as expected? I suspect this happened in https://github.com/scylladb/scylladb/issues/15214.

Reply:
makes sense. since it's an "experimental" option. persumably, our user should not use it. let's go further by dropping it. so scylla can yell at us at seeing this option.

Reply:
#15233